### PR TITLE
Edit and delete

### PR DESCRIPTION
**Pull request by MaksymilianK**

### CODE REVIEW

**Comment by elacymerys** [comment]
It would be great to add some modal with delete confirmation ;-) But we can to it later - not crucial functionality for now

File: thesis-gui/src/services/quizzes-service.ts [hunk 1]
```diff
@@ -8,5 +8,17 @@ export const quizzesService = {
...
+    getQuiz(key: string) {
+        return httpService.get<QuizFullResponse>(`/questions-sets/${key}`);
+    },
+
+    update(key: string, questionsSet: QuestionsSetCreateRequest) {
+        return httpService.put<QuestionsSetCreateRequest, QuestionsSetKeyResponse>(`/questions-sets/${key}`, questionsSet);
+    },
+
+    delete(key: string) {
+        return httpService.delete<undefined, QuestionsSetKeyResponse>(`/questions-sets/${key}`, undefined);
```
Comment (elacymerys) [inline thesis-gui/src/services/quizzes-service.ts]: Why response type here is `QuestionsSetKeyResponse` while the return type of `deleteQuestionsSet` method in `TeacherQuestionsSetController` class is void? Shouldn't it be `undefined`?

File: thesis-gui/src/types/my-quiz.ts [hunk 2]
```diff
@@ -24,3 +24,14 @@ export type QuestionsSetCreateRequest = {
...
     questionsSetKey: string
 }
+
+export type TeacherQuestionResponse = {
+    question: string,
+    correct: string,
+    answers: string[]
+}
+
+export type QuizFullResponse = {
```
Comment (elacymerys) [inline thesis-gui/src/types/my-quiz.ts]: I would rename this `QuizFullResponse` type to `QuestionsSetResponse` or sth like that

File: thesis-gui/src/services/quizzes-service.ts [hunk 3]
```diff
@@ -8,5 +8,17 @@ export const quizzesService = {
...
     createNew(questionsSet: QuestionsSetCreateRequest) {
         return httpService.post<QuestionsSetCreateRequest, QuestionsSetKeyResponse>('/questions-sets', questionsSet);
+    },
+
+    getQuiz(key: string) {
+        return httpService.get<QuizFullResponse>(`/questions-sets/${key}`);
+    },
+
+    update(key: string, questionsSet: QuestionsSetCreateRequest) {
+        return httpService.put<QuestionsSetCreateRequest, QuestionsSetKeyResponse>(`/questions-sets/${key}`, questionsSet);
```
Comment (elacymerys) [inline thesis-gui/src/services/quizzes-service.ts]: I would change this `QuestionsSetCreateRequest` type to `QuestionsSetRequest` or sth like that as now it's used in `createNew` as well as `update` methods

File: thesis-gui/src/components/my-quizzes/MyQuizzes.tsx [hunk 4]
```diff
@@ -1,14 +1,14 @@
...
+    IonButton, IonButtons,
     IonContent, IonIcon,
     IonItem,
     IonLabel,
     IonList,
     IonPage, IonText, useIonToast
 } from "@ionic/react";
 import {useHistory} from "react-router";
-import {keyOutline} from "ionicons/icons";
+import {keyOutline, create, trashBin} from "ionicons/icons";
```
Comment (elacymerys) [inline thesis-gui/src/components/my-quizzes/MyQuizzes.tsx]: I would use the outlined icons ;-) For me looks better but it's only my opinion ;-)

**Review by elacymerys — CHANGES_REQUESTED** [review]

File: thesis-gui/src/services/quizzes-service.ts (hunk 1 above)
Comment (MaksymilianK) [reply thesis-gui/src/services/quizzes-service.ts]: You're right. Changed to `undefined`.

File: thesis-gui/src/types/my-quiz.ts (hunk 2 above)
Comment (MaksymilianK) [reply thesis-gui/src/types/my-quiz.ts]: Done.

File: thesis-gui/src/services/quizzes-service.ts (hunk 3 above)
Comment (MaksymilianK) [reply thesis-gui/src/services/quizzes-service.ts]: Done.

File: thesis-gui/src/components/my-quizzes/MyQuizzes.tsx (hunk 4 above)
Comment (MaksymilianK) [reply thesis-gui/src/components/my-quizzes/MyQuizzes.tsx]: Why not :) Done.

**Review by elacymerys — DISMISSED** [review]

File: thesis-gui/src/components/my-quizzes/MyQuizzes.tsx [hunk 5]
```diff
@@ -29,22 +29,40 @@ const QuizzesListItem: React.FC<{
...
+
+    const handleRefresh = () => {
+        quizzesService.refreshKey({ questionsSetKey: questionsSetKey })
+            .then(res => {
+                copyKeyToClipboard(res.questionsSetKey, REFRESH_KEY_TOAST_MESSAGE);
+                reloadQuizzes();
+            })
+            .catch(err => {
+                if (isApiError(err) && (err as ApiError).apiStatusCode === HttpStatusCode.UNAUTHORIZED) {
+                    tryRefreshTokens().then(handleDelete);
```
Comment (MaksymilianK) [inline thesis-gui/src/components/my-quizzes/MyQuizzes.tsx]: `handleDelete` ----> `handleRefresh`

File: thesis-gui/src/components/my-quizzes/MyQuizzes.tsx [hunk 6]
```diff
@@ -29,22 +29,40 @@ const QuizzesListItem: React.FC<{
...
             duration: 1500,
-            position: 'bottom'
+            position: 'top'
         });
     };
 
-    const copyKeyToClipboard = () => {
-        navigator.clipboard.writeText(questionsSetKey);
-        presentToast();
+    const copyKeyToClipboard = (key: string, toastMessage: string) => {
```
Comment (MaksymilianK) [inline thesis-gui/src/components/my-quizzes/MyQuizzes.tsx:48]: I started to wonder if the key should be displayed somewhere. Currently, the only way to see the key is to copy it and paste somewhere. What if the teacher wants to see the key in order to write it down on the board or something, so the students can see it? But it's not something that has to be done in this PR.

Comment (elacymerys) [reply thesis-gui/src/components/my-quizzes/MyQuizzes.tsx:48]: I was also wandering about that but didn't find any good solution for that

File: thesis-gui/src/components/my-quizzes/MyQuizzes.tsx (hunk 5 above)
Comment (elacymerys) [reply thesis-gui/src/components/my-quizzes/MyQuizzes.tsx]: Done

**Review by elacymerys — APPROVED** [review]

**Pull request merged**